plugins/reporters/freemarker/src/main/kotlin/FreemarkerTemplateProcessor.kt

```diff
@@ -325,6 +330,22 @@ class FreemarkerTemplateProcessor(
         fun collectLicenses(snippetFindings: Collection<SnippetFinding>) : Set<String> =
             snippetFindings.map { it.snippet.licenses.toString() }.toSet()
 
+        /**
+         * Filter the scan results to remove the ones having a VCS provenance without an empty path: Since FossID does
+         * not support support VCS path, they are most likely duplicates of other results.
+         */
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun filterResultsByVCS(scanResults: Map<Identifier, List<ScanResult>>): Map<Identifier, List<ScanResult>> {
```

Comment:
Shouldn't this filtering already happen in the FossID scanner implementation, so such results do not show up in the first place?

Reply:
Hmm I will try to better explain: the project I am scanning contains two packages with the same provenance, differing in the VCS path.
The current scanner logic in ORT sees that these packages have the same provenance, and triggers a scan for one of them in FossID ("Which one" could be influenced thanks to https://github.com/oss-review-toolkit/ort/pull/7092).

Then, after the scan, there is a scanner logic that de duplicates the scan results for packages with the same provenance. Therefore the scan results contains the FossID results two times, one for each package.

Therefore there is no filtering to be done in FossID : the scanner scans already only one package.
I think the whole duplication behavior will be fixed when FossID will scan by provenance and that scan results are stored by provenance.

Reply:
> there is a scanner logic that de duplicates the scan results for packages with the same provenance

Did you mean "duplicates"  instead of "de duplicates"? (I believe duplicating is what it actually does)

> Therefore there is no filtering to be done in FossID : the scanner scans already only one package.

I don't understand how the amount of packages being scanned is related to filtering findings by path. 
To me these topics seem unrelated, maybe there is a mis-understanding here. 

I believe it'd be worth looking at the functions `ScanSummary.filterByPath()` and `ScanSummary.filterByIgnorePatterns()`. These do not adhere to snippet findings yet. I believe they need to be extended, which IIUC would address @sschuberth 's ask.

Reply:
In the Analyzer result, there are two packages with nearly the same provenance: it only differs by `vcs_info/path`. 

When the scanner is called, one package is chosen (=reference package) and scanned with FossID. The other package is NOT scanned. I believe this logic scans the reference package:
https://github.com/oss-review-toolkit/ort/blob/40eb0f0b0ac4709adf0c1ecfbea2a2c1136f04c0/scanner/src/main/kotlin/Scanner.kt#L291-L297

Then after the scan, the `ScanSummary` of the reference package is duplicated to the other package.
At the end, the scan-results contains the two packages having each identical summaries.

The reporter cannot be called on this as the two duplicated summaries are the same. Therefore, this function filters the packages in the scan-results to keep only one.

Reply:
> The reporter cannot be called on this as the two duplicated summaries are the same. Therefore, this function filters the packages in the scan-results to keep only one.

Shouldn't each package have the snippet findings associated with it, no matter if there is another package which includes the same snippets? If both packages are built from same VCS, then adjusting these filterBy functions I mentioned earlier would take care of filtering out the snippets which are not relevant.

Reply:
But I don't want to filter out individual snippet findings: I want to have in the report only ONE set of snippet findings.
Therefore, I need to do filtering at package level to keep only one package.

How is the webapp reporter dealing with duplicated license findings with packages having the same VCS ?

Reply:
> How is the webapp reporter dealing with duplicated license findings with packages having the same VCS ?

AFAIK, the WebApp / static HTML report do report the licenses per package. If two packages are built from same VCS, they still report per package.

Reply:
Following this discussion, I realized this change was controversial so I removed the related commit to not hinder the current PR. I will rework it and add it to a future separate PR.